transcoder.go

```diff
@@ -79,56 +79,60 @@ func (t *Transcoder) ServeHTTP(writer http.ResponseWriter, request *http.Request
 	op.handle()
 }
 
-func (t *Transcoder) registerService(svc *Service, svcOpts serviceOptions) error {
+func (t *Transcoder) registerService(svc *Service, svcOpts serviceOptions) (*serviceOptions, error) {
```

Comment:
nit: mutating the svcOpts and returning the pointer should be done via a constructor.